ccl/changefeedccl: TestChangefeedPrimaryKeyChangeWorks failed

ccl/changefeedccl.TestChangefeedPrimaryKeyChangeWorks [failed](https://tanzanite.cluster.engflow.com/invocations/default/fa614228-f8fe-4e0c-99f1-fcd3dd9c8417?testReportRun=22&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):


```
    changefeed_test.go:6656: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedPrimaryKeyChangeWorks4149107615
--- FAIL: TestChangefeedPrimaryKeyChangeWorks (11.30s)
=== RUN   TestChangefeedPrimaryKeyChangeWorks/enterprise
    helpers_test.go:883: making server as secondary tenant
    helpers_test.go:962: making enterprise feed factory
    helpers_test.go:976: pgURL enterprise root
    changefeed_test.go:6646: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:198
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:273
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6646
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1134
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1169
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:242
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:201
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:266
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:262
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:199
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:273
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedPrimaryKeyChangeWorks.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6646
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamed.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1134
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1169
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) expected
        	            	  |   foo: [14, "e"]->{"after": {"a": 5, "b": "e"}}
        	            	  |   foo: [6, "a"]->{"after": {"a": 6, "b": "a"}}
        	            	  | got
        	            	  |   foo: [1]->{"after": {"a": 1, "b": "f"}}
        	            	  |   foo: [6, "a"]->{"after": {"a": 6, "b": "a"}}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestChangefeedPrimaryKeyChangeWorks/enterprise
    --- FAIL: TestChangefeedPrimaryKeyChangeWorks/enterprise (11.09s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedPrimaryKeyChangeWorks.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-43436

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -6646,8 +6646,6 @@ INSERT INTO foo VALUES (1, 'f');
 		assertPayloads(t, foo, []string{
 			`foo: [6, "a"]->{"after": {"a": 6, "b": "a"}}`,
 			`foo: [14, "e"]->{"after": {"a": 5, "b": "e"}}`,
-		})
-		assertPayloads(t, foo, []string{
 			`foo: [1]->{"after": {"a": 1, "b": "f"}}`,
 		})
 	}

